epoxy-processor/src/main/java/com/airbnb/epoxy/GeneratedModelWriter.java

```diff
@@ -215,29 +214,51 @@ private static ParameterizedTypeName getModelClickListenerType(ClassToGenerateIn
       // view click listener of the original model.
 
       String modelClickListenerField = attribute.getModelClickListenerName();
-      preBindBuilder
-          .beginControlFlow("if ($L != null)", modelClickListenerField)
-          .addCode(CodeBlock.of(
-              "super.$L = new $T() {\n"
-                  + "    // Save the original click listener so if it gets changed on\n"
-                  + "    // the generated model this click listener won't be affected\n"
-                  + "    // if it is still bound to a view.\n"
-                  + "    private final $T $L = $T.this.$L;\n"
-                  + "    public void onClick($T v) {\n"
-                  + "    $L.onClick($T.this, object,\n"
-                  + "        holder.getAdapterPosition());\n"
-                  + "    }\n"
-                  + "    public int hashCode() {\n"
-                  + "       // Use the hash of the original click listener so we don't change the\n"
-                  + "       // value by wrapping it with this anonymous click listener\n"
-                  + "       return $L.hashCode();\n"
-                  + "    }\n"
-                  + "  };\n", attribute.getName(), viewClickListenerType,
-              getModelClickListenerType(classInfo),
-              modelClickListenerField, classInfo.getGeneratedName(),
-              modelClickListenerField, viewType, modelClickListenerField,
-              classInfo.getGeneratedName(), modelClickListenerField))
-          .endControlFlow();
+      preBindBuilder.beginControlFlow("if ($L != null)", modelClickListenerField);
+      if (attribute.isPrivate()) {
+        preBindBuilder.addCode(CodeBlock.of(
+            "super.$L(new $T() {\n"
+                + "    // Save the original click listener so if it gets changed on\n"
+                + "    // the generated model this click listener won't be affected\n"
+                + "    // if it is still bound to a view.\n"
+                + "    private final $T $L = $T.this.$L;\n"
+                + "    public void onClick($T v) {\n"
+                + "    $L.onClick($T.this, object,\n"
+                + "        holder.getAdapterPosition());\n"
+                + "    }\n"
+                + "    public int hashCode() {\n"
+                + "       // Use the hash of the original click listener so we don't change the\n"
+                + "       // value by wrapping it with this anonymous click listener\n"
+                + "       return $L.hashCode();\n"
+                + "    }\n"
+                + "  });\n", attribute.setter(), viewClickListenerType,
```

Comment:
is using the setter vs assigning the field directly the only difference? Maybe we can keep the main code block the same, but assign it to a local variable.

Then in an if/else we can either call the super setter or set the super field. That way we could reuse most of the code